tsv_buddy.rb

```diff
@@ -9,9 +9,7 @@ module TsvBuddy
   # take_tsv: converts a String with TSV data into @data
   # parameter: tsv - a String in TSV format
   def take_tsv(tsv)
-    tsv_data = CSV.parse(tsv, TSV_OPTIONS)
-    column_names = tsv_data.shift
-    @data = tsv_data.map { |row| hashify(column_names, row) }
+    @data = CSV.parse(tsv, TSV_OPTIONS.merge(headers: true)).map(&:to_h)
```

Comment:
setting `:headers` to `true` will create a 2d array with first row containing headers (first row of the file) and second row containing the value of the data.